src/sonic-yang-models/yang-models/sonic-mclag.yang

```diff
@@ -0,0 +1,240 @@
+module sonic-mclag {
+	namespace "http://github.com/Azure/sonic-mclag";
+	prefix smclag;
+	yang-version 1.1;
+
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC MCLAG";
+
+	revision 2019-10-01 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-mclag {
+
+        container MCLAG_DOMAIN {
+
+            list MCLAG_DOMAIN_LIST { 
+                key "domain_id";
+                max-elements 1;
+
+                leaf domain_id {
+                     type uint16 {
+                        range "1..4095" {
+                            error-message "MCLAG Domain ID out of range";
+                        }
+                    }
+                }
+
+                leaf source_ip {
+                    type inet:ipv4-address;
+                }
+                leaf peer_ip {
+                    type inet:ipv4-address;
+                }
+                leaf peer_link {
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                    }
+                }
+                leaf keepalive_interval {
+                    type uint16 {
+                        range "1..60" {
+                            error-message "MCLAG Domain keepalive interval out of range";
+                            error-app-tag keepalive_interval-invalid;
+                        }
+                    }
+                    default 1;
+                }
+                leaf session_timeout {
+                    type uint16 {
+                        range "1..3600" {
+                            error-message "MCLAG Domain session timeout out of range";
+                            error-app-tag session_timeout-invalid;
+                        }
+                    }
+                    default 30;
+                }
+                must "(keepalive_interval * 3) <= session_timeout" {
+                        error-message "(keepalive interval * 3) <= session_timeout value";
+                        error-app-tag keepalive_interval-invalid;
+                }
+            }
+        }
```

Comment:
We need the UT cases to cover various conditions present in the YANG.

Reply:
working on adding UT cases. will check in soon

Reply:
added some base case, working on adding some more

Reply:
@venkatmahalingam 
added UT Test cases. thanks